intro_to_R_tidyverse/intro_to_tidyverse.Rmd

```diff
@@ -0,0 +1,173 @@
+special.mtcars <- select(special.mtcars, "mpg", "cyl", "gear")
+```
+UGH, we have to repeat and assign `special.mtcars` so many times here! 
+It's annoying and makes it harder for people to read. 
+
+*Example 2:* Same result as 1 but with pipes!
+```{r}
+# Example of the same modifications as above but with pipes!
+special.mtcars <- mtcars %>% 
+                  rownames_to_column("car_model") %>% 
+                  arrange(mpg) %>%
+                  filter(mpg > 18) %>% 
+                  select("mpg", "cyl", "gear")
+```
+
+Now that hopefully you are convinced that tidyverse helps your code to be neater and
+easier to use and read, let's go through some of the popular tidyverse functions
+and so we can create pipelines like this. 
+
+## Common tidyverse functions
+
+Let's say we wanted to filter this gene expression dataset to particular sample
+groups.
+In order to do this, we would use the function `filter` as well as a logic statement.
+```{r}
+# Here let's filter the data to have only females
+metadata %>% filter(sex == "F")
+```
+
+We can use filter similarly for numeric statements.  
+```{r}
+# Here let's filter the data to have people only older than 
+metadata %>% filter(age_mths > 100)
+```
+
+What if we wanted filter this dataset to males who are younger than 100 months? 
+Let's make a longer pipe.
+```{r }
+metadata %>% filter(age_mths > 100) %>%
+             filter(sex == "M")
+```
+
+Let's build on to this pipe and keep only some of these variables with the 
+`select` function. 
+```{r}
+metadata %>% filter(age_mths > 100) %>%
+             filter(sex == "M") %>% 
+             select(sample_id, histology, metastasis)
+```
+
+Let's say we wanted to arrange this dataset so that the patients were arranged
+youngest to oldest.
+In order to do this, we would use the function `arrange` as well as the variable
+we would like to sort by (in this case `age_mths`).
+```{r}
+metadata %>% arrange(age_mths) 
+```
+
+What if we want the the oldest patients at the top of this dataset instead? 
+We can use the same function, but instead use the `desc` function too. 
+```{r}
+metadata %>% arrange(desc(age_mths))
+```
+
+What if we want to obtain summary statistics based on groups? 
+Conveniently named, there's a function called `group_by` that seamlessly allows
+us to do this. 
+Also note that you can group by multiple variables at a time if you want to.
+```{r}
+metadata %>% group_by(sex) %>% 
+             summarize(mean(age_mths))
+```
+What if we would like to create new variables based on other variables? 
+For that we use `mutate` function. 
+Let's convert the ages that are in months to years and make that into a new variable.
+Here we are also using the `round` function from base R to limit the number of 
+decimal points in our new variable to 2.
+```{r}
+metadata %>% mutate("age_yrs" = round(age_mths/12, 2)) %>%
+             select(sample_id, age_mths, age_yrs)
+```
+
+## Gene data examples
```

Comment:
I don't know that this section really adds too much over the metadata section (that we don't cover elsewhere) and has potential to be confusing.

Reply:
I think it might be useful to see how one could also work with gene expression data.